python/templates/Collection.h.jinja2

```diff
@@ -61,8 +61,12 @@ public:
   /// number of elements in the collection
   size_t size() const override final;
 
+  /// fully qualified type name
+  std::string getTypeName() const override { return std::string("{{ (class | string ).strip(':')+"Collection" }}"); }
   /// fully qualified type name of elements - with namespace
   std::string getValueTypeName() const override { return std::string("{{ (class | string ).strip(':') }}"); }
+  /// fully qualified type name of stored POD elements - with namespace
+  std::string getDataTypeName() const override { return std::string("{{ (class | string ).strip(':')+"Data" }}"); }
```

Comment:
It looks like this will always be used like this
```cpp
const auto bufferClassName = "std::vector<" + coll->getDataTypeName() + ">";
```
Should we simply make this return the full thing (and potentially give it a name that indicates that)? Or do we foresee any potential use cases where the `Data` name alone would be useful?

Reply:
In a sense the fact that a vector<> is used here is an implementation detail of the ROOTReader/Writer and other future implementations (HDF5 ?) might make a different use of the underlying POD data type. 
This is why I thought it would be more generic this way...